components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx

```diff
@@ -414,21 +416,41 @@ export class Panel extends React.Component<Props, State> {
     return defaultContribution
   }
 
+  initiateDelayCounter = () => {
```

Comment:
is this delay fake? so it doesn't reflect actual response from c++?

What happens if connection is really slow and it takes 15s to get publisher list from the server and result from c++ into JS?

Reply:
Progress spinner is shown for minimum of 2 seconds. Can show for longer but this ensures that user at least sees the spinner.